common/src/tracer.rs

```diff
@@ -23,20 +24,102 @@ macro_rules! conditional_init_tracer {
     ($otel_endpoint:expr, $level: expr) => {
         $otel_endpoint
             .map(|otel_endpoint| {
-                OtelTracer::new(
+                let tracer = OtelTracer::new(
                     otel_endpoint,
                     env!("CARGO_BIN_NAME"),
                     Some((module_path!(), $level)),
-                )
-                .expect("Open Telemetry Tracer is created")
+                );
             })
             .or_else(|| {
                 tracing_subscriber::fmt::init();
```

Comment:
Should there be a warning here that OpenTelemetry was specified but unavailable?

Reply:
Why not here? You are initialising `tracing_subscriber` which I assume you are only doing because OpenTelemetry is not available, if not here then where?

Reply:
The `or_else` is only called if the user deliberately does not provided a otel-endpoint cli parameter. These lines are not error handling.